lib/fpm/package/p5p.rb

```diff
@@ -0,0 +1,116 @@
+require "erb"
+require "fpm/namespace"
+require "fpm/package"
+require "fpm/errors"
+require "fpm/util"
+
+class FPM::Package::P5P < FPM::Package
+
+  option "--user", "USER",
+    "Set the user to USER in the prototype files.",
+    :default => 'root'
+
+  option "--group", "GROUP",
+    "Set the group to GROUP in the prototype file.",
+    :default => 'root'
+
+  option "--zonetype", "ZONETYPE",
+    "Set the allowed zone types (global, nonglobal, both)",
+    :default => 'value=global value=nonglobal' do |value|
+      case @value
+      when "both"
+        value = "value=global value=nonglobal"
+      else
+        value = "value=#{value}"
+      end
+    end # value
+
+  option "--publisher", "PUBLISHER", 
+    "Set the publisher name for the repository",
+    :default => 'FPM'
+
+  option "--lint" , :flag, "Check manifest with pkglint",
+    :default => true
+
+  option "--validate", :flag, "Validate with pkg install",
+    :default => true
+
+  def architecture
+    case @architecture
+    when nil, "native"
+      @architecture = %x{uname -p}.chomp
+    when "all"
+      @architecture = 'i386 value=sparc'
+    end
+
+    return @architecture
+  end # def architecture
+
+  def output(output_path)
+    
+    # Fixup the category to an acceptable solaris category
+    case @category
+    when nil, "default"
+      @category = 'Applications/System Utilities'
+    end
+
+    # Generate a package manifest.
+    pkg_generate = safesystemout("pkgsend", "generate", "#{staging_path}")
+    File.open("#{build_path}/#{name}.p5m.1", "w") do |manifest|
```

Comment:
For all these `"#{build_path}/...` examples, the `build_path` method lets you pass an argument for what to append to it. The goal was to avoid having tons of string evaluations `"#{build_path}/whatever"` which you can do instead as `build_path(whatever)`.

In this example, rewriting:

```ruby
File.open("#{build_path}/#{name}.p5m.1", "w")
```

As this:

```ruby
File.open(build_path("{name}.p5m.1"), "w")
```

It's intended that the 2nd way is more readable, but readability is open to interpretation, so this is only a suggestion :)